comp/core/autodiscovery/scheduler/controller.go

```diff
@@ -43,9 +38,16 @@ type Controller struct {
 	stopChannel chan struct{}
 }
 
-// NewController inits a scheduler controller
+// NewController inits a scheduler controller without waiting, for unit test only
 func NewController() *Controller {
```

Comment:
It might have been better to keep only one function to create the Controller and have an option to start it, something like:
```
func NewController(start bool) *Controller {
```